eth/blockwatch/block_watcher.go

```diff
@@ -146,6 +146,25 @@ func (w *Watcher) InspectRetainedBlocks() ([]*MiniHeader, error) {
 	return w.stack.Inspect()
 }
 
+func (w *Watcher) syncToLatestBlock() error {
```

Comment:
Is it possible for issues to arise if multiple calls for this method are in-progress? i.e. After tick 1, we start syncing N blocks, but before that sync completes, tick 2 occurs and we start another sync. Seems like this would depend on the polling interval (which sets how often ticks/syncs occur) and the speed of the header fetching requests - likely not a common issue, but maybe possible to occur under certain circumstances?

I see that the upstream package uses a [mutex](https://github.com/0xProject/0x-mesh/blob/80d4c33b9c753dd967dd653d6fbfb8f46dffc606/ethereum/blockwatch/block_watcher.go#L92) that might guard against this particular issue.

Reply:
fixed in https://github.com/livepeer/go-livepeer/pull/1729/commits/511c85802c591398ed2ba931af5013a70cea502a

Reply:
Forgot `defer` for the unlock added in subsequent commit